official/utils/arg_parsers/parsers.py

```diff
@@ -58,9 +58,36 @@ def __init__(self):
 from __future__ import division
 from __future__ import print_function
 
-
 import argparse
 
+import tensorflow as tf
+
+
+def parse_dtype_info(flags):
+  """Convert dtype string to tf dtype, and set loss_scale default as needed.
+
+  Args:
+    flags: namespace object returned by arg parser.
+
+  Raises:
+    ValueError: If an invalid dtype is provided.
+  """
+  flags.dtype = {
+      "fp16": tf.float16,
+      "float16": tf.float16,
+      "fp32": tf.float32,
+      "float32": tf.float32,
+  }.get(flags.dtype, flags.dtype)
+
+  if flags.dtype is None or isinstance(flags.dtype, str):
+    raise ValueError("Invalid dtype: {}".format(flags.dtype))
+
+  if flags.loss_scale is None:
+    flags.loss_scale = {
+        "float16": 128,
+        "float32": 1,
+    }[flags.dtype.name]
```

Comment:
Also a constant? Perhaps in the same DTYPE_MAP. Ideally keyed on the same args, rather than a different string drawn from the TF name, which is not in our control.